pallets/moonbeam-lazy-migrations/src/lib.rs

```diff
@@ -106,5 +121,47 @@ pub mod pallet {
 			)
 			.into())
 		}
+
+		// TODO(rodrigo): This extrinsic should be removed once the storage of destroyed contracts
+		// has been removed
+		#[pallet::call_index(1)]
+		#[pallet::weight({
+			let addresses_len = addresses.len() as u32;
+			<T as crate::Config>::WeightInfo::clear_suicided_storage(addresses_len, *limit)
+		})]
+		pub fn clear_suicided_storage(
+			origin: OriginFor<T>,
+			addresses: BoundedVec<H160, GetArrayLimit>,
+			limit: u32,
+		) -> DispatchResultWithPostInfo {
+			ensure_signed(origin)?;
+
+			ensure!(limit != 0, Error::<T>::LengthOrLimitCannotBeZero);
+			ensure!(addresses.len() != 0, Error::<T>::LengthOrLimitCannotBeZero);
+
+			let mut limit = limit as usize;
+
+			for address in &addresses {
+				// Ensure that the contract is suicided by checking that it has no code and at least
+				// one storage entry.
+				ensure!(
+					!pallet_evm::AccountCodes::<T>::contains_key(&address)
+						&& pallet_evm::AccountStorages::<T>::iter_key_prefix(&address)
```

Comment:
We should also ensure that the contract is not marked as Suicided on the new storage item pallet_evm::Suicided, because if it is, it should not be clear by this lazy migration